auditbeat/module/file_integrity/eventreader_fsnotify.go

```diff
@@ -39,19 +39,19 @@ type reader struct {
 
 // NewEventReader creates a new EventProducer backed by fsnotify.
 func NewEventReader(c Config) (EventProducer, error) {
-	watcher, err := monitor.New(c.Recursive)
```

Comment:
I wonder if this was here to detect config mistakes early, @andrewkroh probably you can confirm

Reply:
I don't think much validation is gained with this particular instantiation from a config validation perspective. So moving it into `Start` should be OK.